src/main/java/com/iota/iri/cache/impl/CacheImpl.java

```diff
@@ -0,0 +1,102 @@
+package com.iota.iri.cache.impl;
+
+import com.iota.iri.cache.Cache;
+import com.iota.iri.cache.CacheConfiguration;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import com.google.common.collect.MapMaker;
+
+/**
+ * Cache operations
+ */
+public class CacheImpl<K, V> implements Cache<K, V> {
+
+    /**
+     * Cache config
+     */
+    private final CacheConfiguration cacheConfiguration;
+
+    /**
+     * Store the state of entries. True for "Fresh" and False for "dirty"
+     */
+    private final ConcurrentMap<K, Boolean> entryState;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused private fields such as 'entryState'.](https://app.codacy.com/app/GalRogozinski/iri/pullRequest?prid=4567264)

Reply:
What do we do about these `unused` fields since we are not implementing yet?

Reply:
It is fine, since it will be resolved later